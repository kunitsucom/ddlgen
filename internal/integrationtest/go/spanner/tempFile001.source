package main

type (
	// User is a user.
	//
	// spanddl:      table: `Users`
	// spanddl:     option: PRIMARY KEY (`Id`)
	// spanddl: constraint: CONSTRAINT AgeGTEZero CHECK(Age >= 0)
	// spanddl:      index: CREATE INDEX `IndexUsersByName` ON Users(`Name`)
	User struct {
		// ID is a user ID.
		ID string    `db:"Id"    spanddl:"STRING(36)  NOT NULL"`
		// Name is a user name.
		Name string  `db:"Name"  spanddl:"STRING(255) NOT NULL"`
		// Email is a user email.
		Email string `db:"Email" spanddl:"STRING(255) NOT NULL"`
		// Age is a user age.
		Age int      `db:"Age"   spanddl:"INT64       NOT NULL"`
	}
	// Book is a book.
	//
	// spanddl:table:`Books`
	// spanddl:options:PRIMARY KEY (`Id`)
	Book struct {
		// ID is a book ID.
		ID string     `db:"Id"     spanddl:"STRING(36)  NOT NULL"`
		// Title is a book title.
		Title string  `db:"Title"  spanddl:"STRING(255) NOT NULL"`
		// Author is a book author.
		Author string `db:"Author" spanddl:"STRING(255) NOT NULL"`
	}
)
