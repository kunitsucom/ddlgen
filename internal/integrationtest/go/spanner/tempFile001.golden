-- Code generated by ddlgen. DO NOT EDIT.
--
-- Date: 2021-01-01T09:00:00+09:00
--

-- source: spanner/tempFile001.source:9
-- User is a user.
--
-- spanddl:table:`Users`
-- spanddl:option:PRIMARY KEY (`Id`)
-- spanddl:constraint:CONSTRAINT AgeGTEZero CHECK(Age >= 0)
CREATE TABLE `Users` (
    -- ID is a user ID.
    `ID`    STRING(36)  NOT NULL,
    -- Name is a user name.
    `Name`  STRING(255) NOT NULL,
    -- Email is a user email.
    `Email` STRING(255) NOT NULL,
    -- Age is a user age.
    `Age`   INT64       NOT NULL,
    CONSTRAINT AgeGTEZero CHECK(Age >= 0)
) PRIMARY KEY (`Id`);

-- source: spanner/tempFile001.source:23
-- Book is a book.
--
-- spanddl:table:`Books`
-- spanddl:options:PRIMARY KEY (`Id`)
CREATE TABLE `Books` (
    -- ID is a book ID.
    `ID`     STRING(36)  NOT NULL,
    -- Title is a book title.
    `Title`  STRING(255) NOT NULL,
    -- Author is a book author.
    `Author` STRING(255) NOT NULL
) PRIMARY KEY (`Id`);
